Added new introductory validation notebooks

validbeck opened the pull request:
## Internal Notes for Reviewers

> sc-9191

[**LIVE PREVIEW**](https://docs-demo.vm.validmind.ai/pr_previews/beck/sc-9191/add-new-validmind-for-model-validation-notebooks/developer/validmind-library.html)

### ValidMind for model validation notebooks

These were added to the "developer" (now library) portal:

| Sidenav | Listing tiles |
|---|---|
|<img width="363" alt="Screenshot 2025-04-09 at 2 36 33 PM" src="https://github.com/user-attachments/assets/f286ee8d-3936-4d3d-bb74-2c13a6f44f4b" /> | <img width="795" alt="Screenshot 2025-04-09 at 2 36 29 PM" src="https://github.com/user-attachments/assets/61e23af0-3602-4534-b807-4086a71021af" />|

### ValidMind for model development notebooks

These have the updated numbering structure:

| Sidenav | Listing tiles |
|---|---|
|<img width="361" alt="Screenshot 2025-04-09 at 2 36 23 PM" src="https://github.com/user-attachments/assets/dfeb49d8-62de-4e8f-91ea-604306421f74" /> | <img width="819" alt="Screenshot 2025-04-09 at 2 36 19 PM" src="https://github.com/user-attachments/assets/9c8f0a36-43b8-4198-89d6-4006b311be08" />|

### Developers > ValidMind Library

- Since this portal is much more than a developer portal, I've made the language and links surrounding the portal more universal:

| Main drop-down | Collapsed drop-down |
|---|---|
| <img width="453" alt="Screenshot 2025-04-09 at 2 35 39 PM" src="https://github.com/user-attachments/assets/a7b9c0da-e793-40d4-9479-063d0623e16d" /> | <img width="292" alt="Screenshot 2025-04-09 at 2 34 21 PM" src="https://github.com/user-attachments/assets/d7071541-64fb-4f78-8455-fec350781843" />|

- You may notice that I also applied the styling that the API docs have to this and the notebooks/test descriptions, since it's jarring for the two sections to have different side navigations (BONUS: This also takes care of the really squished long-titles we have for notebooks!):

<img width="1706" alt="Screenshot 2025-04-09 at 2 37 38 PM" src="https://github.com/user-attachments/assets/c662fb02-32e8-4d0b-b8bf-83d606671b6c" />


- I also moved the sidebar for this landing section to its own `_sidebar.yaml`:

https://github.com/validmind/documentation/blob/a89901c1e08dd3320d989474289e0cc03b652c64/site/developer/_sidebar.yaml#L3

https://github.com/validmind/documentation/blob/a89901c1e08dd3320d989474289e0cc03b652c64/site/_quarto.yml#L5

#### ValidMind Library API

There is a sibling PR to this to change the reference to include the API in the title/link for that landing page as this was somehow missed in the initial one — again, trying to reduce confusion with our habit of naming pages/sections the same darn thing!

**https://github.com/validmind/validmind-library/pull/350**

<img width="1728" alt="Screenshot 2025-04-09 at 2 35 20 PM" src="https://github.com/user-attachments/assets/e9b7c20b-0b3b-4458-8fc1-d2e10ac12fc9" />

#### Introduction for model developers

This old notebook and its executed version I reverted to `main` (`documentation` side only) as we haven't yet updated our training to be able to get rid of the notebook — working on it. 

### validmind.com

I removed this link from the main top navigation as it doesn't need to feature so heavily when it's also in the footer, based on discussions we had during the homepage redesign. 

| Main nav | Collapsed nav |
|---|---|
| <img width="1710" alt="Screenshot 2025-04-09 at 2 34 16 PM" src="https://github.com/user-attachments/assets/9b7ffe00-0134-4e7f-9b27-13a2c56143b0" /> | <img width="292" alt="Screenshot 2025-04-09 at 2 34 21 PM" src="https://github.com/user-attachments/assets/b0b03b47-9665-45d4-8ff3-7af55636d483" />|

nrichers commented on site/_quarto.yml:
```diff
@@ -89,17 +90,17 @@ website:
           file: support/support.qmd
         - text: "---"
-        - text:  "{{< fa cube >}} Developers"
+        - text:  "{{< fa cube >}} Python Library"
```
This is another term we don't actually have in `_variables.yml`. The Python Library is not synonymous with the ValidMind Library and so just "Python Library" to me sounds a bit odd, as it's reductionist for our developer tools which are a collection of thing.

Another way of saying this is that "ValidMind Library" is the parent umbrella term under which all the other various bits are organized (notebooks, API, etc.) and if we're trying to broaden the appeal of this section beyond just developers I would stick with that term. For the API, I would use "Python API" or "ValidMind Library Python API" but not a combination of other things that don't actually resolve the ambiguity for the future. (And you are entirely right to call out some ambiguity here which is what prompted you to make changes in the first place.)

(I'm going to hold off on adding more comments until we have a chance to chat at our standup. 😁 )

validbeck replied:
We have the `Python Library API` in the variables but it didn't quite fit here ...

validbeck commented:
Changed references to the API to "ValidMind Library Python API" here as well, and updated the variable to match:

https://github.com/validmind/documentation/blob/db08c001f32adb7535ff7ad3d5b36ef8fd127082/site/_variables.yml#L22

<img width="288" alt="Screenshot 2025-04-09 at 4 34 06 PM" src="https://github.com/user-attachments/assets/929f1978-392d-4837-a024-f53b829cf38c" />
<img width="1726" alt="Screenshot 2025-04-09 at 4 34 18 PM" src="https://github.com/user-attachments/assets/f36d0b3e-401e-4875-a7e9-98849e118202" />
<img width="456" alt="Screenshot 2025-04-09 at 4 34 25 PM" src="https://github.com/user-attachments/assets/ac29dcc5-31cb-4a32-98af-b610528b74bd" />

nrichers reviewed (approved):
LGTM! Great PR. 🥳 

> You may notice that I also applied the styling that the API docs have ... (BONUS: This also takes care of the really squished long-titles we have for notebooks!)

I quite appreciate this change and I agree that the styling is preferable, so thank you. 💖 

We also have an issue with long sidebar entries in the rest of the docs site that frequently cause wrapping. I would be amenable to applying the same styling from the API reference elsewhere, so it is consistent throughout the site. (Not in this PR, but as a future improvement.)

nrichers commented:
@validbeck a similar story here — while debugging my new script to generate release notes, I was trying to figure out why it was skipping docs updates. Not only are these changes not internal, they are should likely also be labelled with `highlight`.

validbeck commented:
cc: @nrichers 

These should be surfaced in the release notes alongside the actual release for the notebooks. Please refer to the external release notes on the notebook PR here: 

https://github.com/validmind/validmind-library/pull/348
